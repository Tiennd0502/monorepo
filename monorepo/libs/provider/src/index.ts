export { default as TamaguiProvider } from './TamaguiProvider';
export { default as QueryClientProvider } from './QueryClientProvider';
