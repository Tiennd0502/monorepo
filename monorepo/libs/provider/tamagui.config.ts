import { createTamagui, createTokens } from 'tamagui';
import { config } from '@tamagui/config/v3';
import { themes } from '@tamagui/themes';

import {
  colorsMapping,
  radius,
  size,
  space,
  zIndex,
  defaultFont,
  headingFont,
} from '@monorepo/ui';

const tokens = {
  radius,
  size,
  space,
  zIndex,
};

const tamaguiConfig = createTamagui({
  ...config,
  defaultFont: 'body',
  themes,
  fonts: {
    body: defaultFont,
    heading: headingFont,
  },
  tokens: createTokens({
    ...tokens,
    color: colorsMapping,
  }),
});

type AppConfig = typeof config;

declare module 'tamagui' {
  type TamaguiCustomConfig = AppConfig;
}

export default tamaguiConfig;
