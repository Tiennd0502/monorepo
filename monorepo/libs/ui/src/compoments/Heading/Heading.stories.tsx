import { Meta, StoryObj } from '@storybook/react';

import Heading from '.';

const meta: Meta<typeof Heading> = {
  title: 'Components/Heading',
  component: Heading,
};

export default meta;

type Story = StoryObj<typeof Heading>;

export const Default: Story = {
  args: {
    children: 'Default heading',
  },
};

export const Large: Story = {
  args: {
    children: 'Heading large',
    size: 'large',
  },
};

export const ExtraLarge: Story = {
  args: {
    children: 'Heading extraLarge',
    size: 'extraLarge',
  },
};
