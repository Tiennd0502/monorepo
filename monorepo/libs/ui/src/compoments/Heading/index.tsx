import { Heading as THeading, styled } from 'tamagui';

const Heading = styled(THeading, {
  color: '$textBold',
  fontFamily: '$body',
  fontWeight: 'bold',
  allowFontScaling: false,

  variants: {
    title: {
      true: {
        fontFamily: '$title',
      },
    },

    header: {
      true: {
        fontFamily: '$heading',
      },
    },

    size: {
      small: {
        fontSize: '$4',
        lineHeight: '$small',
      },

      medium: {
        fontSize: '$4.5',
        lineHeight: '$medium',
      },

      large: {
        fontSize: '$6',
        lineHeight: '$medium',
      },

      extraLarge: {
        fontSize: '$7.5',
        lineHeight: '$extraLarge',
      },

      huge: {
        fontSize: '$9',
        lineHeight: '$huge',
      },
    },
  } as const,

  defaultVariants: {
    title: true,
    size: 'medium',
  },
});

export default Heading;
