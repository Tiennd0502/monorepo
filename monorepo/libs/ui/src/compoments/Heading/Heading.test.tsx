import {render} from '@testing-library/react-native';

import { TamaguiProvider } from '@monorepo/provider';
import Heading from '.';

const baseProps = {
  children: 'Heading text'
};

const setup = (moreProps = {}) => {
  const props = {
    ...baseProps,
    ...moreProps,
  };
  
  return render(
    <TamaguiProvider>
      <Heading {...props} />
    </TamaguiProvider>
  );
};

describe('Heading', () => {
  it('should render Heading component', () => {
    const container = setup();

    expect(container).toMatchSnapshot();
  });

  it('should render Heading component with size large', () => {
    const container = setup({size: 'large'});

    expect(container).toMatchSnapshot();
  });
});
