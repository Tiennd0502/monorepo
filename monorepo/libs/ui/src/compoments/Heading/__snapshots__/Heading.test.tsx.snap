// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should render Heading component 1`] = `
<Text
  accessibilityRole="header"
  allowFontScaling={false}
  selectable={true}
  style={
    {
      "color": "#303030",
      "fontFamily": "Gelasio-Bold",
      "fontSize": 18,
      "lineHeight": 28,
      "marginBottom": 0,
      "marginLeft": 0,
      "marginRight": 0,
      "marginTop": 0,
    }
  }
  suppressHighlighting={true}
>
  Heading text
</Text>
`;

exports[`Heading should render Heading component with size large 1`] = `
<Text
  accessibilityRole="header"
  allowFontScaling={false}
  selectable={true}
  style={
    {
      "color": "#303030",
      "fontFamily": "Gelasio-Bold",
      "fontSize": 24,
      "lineHeight": 28,
      "marginBottom": 0,
      "marginLeft": 0,
      "marginRight": 0,
      "marginTop": 0,
    }
  }
  suppressHighlighting={true}
>
  Heading text
</Text>
`;
